React/Modules/RCTAlertManager.m

```diff
@@ -64,7 +64,7 @@ - (void)invalidate
   NSString *message = [RCTConvert NSString:args[@"message"]];
   NSString *type = [RCTConvert NSString:args[@"type"]];
   NSDictionaryArray *buttons = [RCTConvert NSDictionaryArray:args[@"buttons"]];
-  BOOL allowsTextInput = [type isEqual:@"plain-text"];
+  BOOL allowsTextInput = [type length] != 0;
```

Comment:
What do you think about replacing `allowsTextInput` with `type.length` instead of adding as the last clause? This way if type is undefined it wont execute the multi if statement as well.